src/main/java/org/traccar/protocol/Vt200ProtocolDecoder.java

```diff
@@ -67,37 +67,46 @@ protected Object decode(
         int type = buf.readUnsignedShort();
         buf.readUnsignedShort(); // length
 
-        if (type == 0x2086 || type == 0x2084 || type == 0x2082) {
+        if (type == 0x2086 || type == 0x2084 || type == 0x2082 || type == 0x3089) {
 
             Position position = new Position(getProtocolName());
             position.setDeviceId(deviceSession.getDeviceId());
 
-            buf.readUnsignedByte(); // data type
+            if (type == 0x3089) {
+                position.set(Position.KEY_IGNITION, buf.readUnsignedByte() == 1);
+            } else {
+                buf.readUnsignedByte(); // data type
+            }
+
             buf.readUnsignedShort(); // trip id
 
             position.setTime(decodeDate(buf));
 
-            position.setLatitude(decodeCoordinate(BcdUtil.readInteger(buf, 8)));
-            position.setLongitude(decodeCoordinate(BcdUtil.readInteger(buf, 9)));
-
-            int flags = buf.readUnsignedByte();
-            position.setValid(BitUtil.check(flags, 0));
-            if (!BitUtil.check(flags, 1)) {
-                position.setLatitude(-position.getLatitude());
-            }
-            if (!BitUtil.check(flags, 2)) {
-                position.setLongitude(-position.getLongitude());
+            if (buf.readableBytes() >= 9) {
```

Comment:
Can we check the `type` here instead?

Reply:
This is there just for the type `0x3089`, as it doesn't return the location every time. For the other types, it always return the location, so this condition will be always positive. I would keep this as it is doing the work.

Reply:
@tananaev, is there any plan to merge this PR?

Reply:
Is the documentation incorrect? From what I can tell, it should always be present.

Reply:
From the documentation, page 58:
```
0x28 0x30 0x14 0x04 0x14 0x00 0x01 0x30 0x89 0x00 0x09 0x01 0x00 0x03 0x28 0x04 0x14 0x09 0x00 0x04 0xDF 0x29  //Acc on report，without GPS info
0x28 0x30 0x14 0x04 0x14 0x00 0x01 0x30 0x89 0x00 0x09 0x00 0x00 0x03 0x28 0x04 0x14 0x09 0x00 0x04 0xDF 0x29  //Acc off report，without GPS info
0x28 0x30 0x14 0x04 0x14 0x00 0x01 0x30 0x89 0x00 0x12 0x01 0x00 0x03 0x28 0x04 0x14 0x09 0x00 0x04 0x22 0x43 0x88 0x16 0x11 0x34 0x91 0x23 0x97 0xDF 0x29  //Acc on report with GPS info
0x28 0x30 0x14 0x04 0x14 0x00 0x01 0x30 0x89 0x00 0x12 0x00 0x00 0x03 0x28 0x04 0x14 0x09 0x00 0x04 0x22 0x43 0x88 0x16 0x11 0x34 0x91 0x23 0x97 0xDF 0x29  //Acc off report with GPS info
```

Reply:
OK, in that case can we replace it with `buf.readableBytes() > 2`?

Reply:
Done. That makes sense.